daemon/config/config_unix_test.go

```diff
@@ -3,21 +3,59 @@
 package config
 
 import (
-	"io/ioutil"
-	"runtime"
-
 	"testing"
+
+	"github.com/docker/docker/opts"
+	"github.com/docker/docker/pkg/testutil/tempfile"
+	"github.com/docker/go-units"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestDaemonConfigurationMerge(t *testing.T) {
-	f, err := ioutil.TempFile("", "docker-config-")
-	if err != nil {
-		t.Fatal(err)
+func TestGetConflictFreeConfiguration(t *testing.T) {
+	configFileData := string([]byte(`
+		{
+			"debug": true,
+			"default-ulimits": {
+				"nofile": {
+					"Name": "nofile",
+					"Hard": 2048,
+					"Soft": 1024
+				}
+			},
+			"log-opts": {
+				"tag": "test_tag"
+			}
+		}`))
+
+	file := tempfile.NewTempFile(t, "docker-config", configFileData)
+	defer file.Remove()
+
+	flags := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	var debug bool
+	flags.BoolVarP(&debug, "debug", "D", false, "")
+	flags.Var(opts.NewNamedUlimitOpt("default-ulimits", nil), "default-ulimit", "")
+	flags.Var(opts.NewNamedMapOpts("log-opts", nil, nil), "log-opt", "")
+
+	cc, err := getConflictFreeConfiguration(file.Name(), flags)
```

Comment:
The change here is not passing a nil parameter for flags. 
The change here can explain:

> Since I am confused that the test here https://github.com/docker/docker/blob/f3a8886d8890d6797a568ab316baf9400ee2e1e1/daemon/config/config_unix_test.go#L23-L29 still work without this PR